mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/FlinkUnifiedCatalog.java

```diff
@@ -0,0 +1,512 @@
+import org.apache.flink.table.catalog.CatalogPartitionSpec;
+import org.apache.flink.table.catalog.ObjectPath;
+import org.apache.flink.table.catalog.exceptions.CatalogException;
+import org.apache.flink.table.catalog.exceptions.DatabaseAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.DatabaseNotEmptyException;
+import org.apache.flink.table.catalog.exceptions.DatabaseNotExistException;
+import org.apache.flink.table.catalog.exceptions.FunctionAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.FunctionNotExistException;
+import org.apache.flink.table.catalog.exceptions.PartitionAlreadyExistsException;
+import org.apache.flink.table.catalog.exceptions.PartitionNotExistException;
+import org.apache.flink.table.catalog.exceptions.PartitionSpecInvalidException;
+import org.apache.flink.table.catalog.exceptions.TableAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotPartitionedException;
+import org.apache.flink.table.catalog.exceptions.TablePartitionedException;
+import org.apache.flink.table.catalog.stats.CatalogColumnStatistics;
+import org.apache.flink.table.catalog.stats.CatalogTableStatistics;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.factories.Factory;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/** This is a Flink catalog wrap a unified catalog. */
+public class FlinkUnifiedCatalog extends AbstractCatalog {
+  private UnifiedCatalog unifiedCatalog;
+  private final String amsUri;
+  private final String amoroCatalogName;
+  /**
+   * Available Flink catalogs for Unified Catalog.
+   *
+   * <p>May include: Iceberg, Mixed and Paimon Catalogs, etc.
+   */
+  private final Map<TableFormat, AbstractCatalog> availableCatalogs;
+
+  public FlinkUnifiedCatalog(
+      String amsUri,
+      String name,
+      String defaultDatabase,
+      Map<TableFormat, AbstractCatalog> availableCatalogs) {
+    super(name, defaultDatabase);
+    this.amsUri = amsUri;
+    this.amoroCatalogName = ArcticThriftUrl.parse(amsUri, THRIFT_TABLE_SERVICE_NAME).catalogName();
+    this.availableCatalogs = availableCatalogs;
+  }
+
+  @Override
+  public void open() throws CatalogException {
+    unifiedCatalog =
+        UnifiedCatalogLoader.loadUnifiedCatalog(amsUri, amoroCatalogName, Maps.newHashMap());
+    availableCatalogs.forEach((tableFormat, catalog) -> catalog.open());
+  }
+
+  @Override
+  public void close() throws CatalogException {
+    if (availableCatalogs != null) {
+      availableCatalogs.forEach((tableFormat, catalog) -> catalog.close());
+    }
+  }
+
+  @Override
+  public List<String> listDatabases() throws CatalogException {
+    return unifiedCatalog.listDatabases();
+  }
```

Comment:
Accoding to review https://github.com/NetEase/amoro/pull/2269#discussion_r1423361566_  

Even if the parent class or interface method declares an exception, if the method itself does not throw an exception, the method signature should not declare an exception.

Reply:
I've fixed it. PTAL